recycleIT-frontend/src/Components/Header/Header.tsx

```diff
@@ -8,33 +8,45 @@ import Avatar from '@mui/material/Avatar';
 import Button from '@mui/material/Button';
 import Tooltip from '@mui/material/Tooltip';
 import MenuItem from '@mui/material/MenuItem';
-import styles from './Header.module.css'
-import leaf from '../assets/leaf.png'
-import AccountCircleFilled from '../assets/AccountCircleFilled.png';
-import LogoutOutlined from '../assets/LogoutOutlined.png'
-import burger from '../assets/burger.png'
-import { Link } from 'react-router-dom';
+import styles from './Header.module.css';
+import MenuIcon from '@mui/icons-material/Menu';
+import IconLogo from './../assets/IconLogo.svg';
+import { Link, NavLink, useNavigate, useLocation } from 'react-router-dom';
+import { UserContext } from '../UserContext/UserContextProvider';
+import { AccountCircle, Logout } from '@mui/icons-material';
+import CloseIcon from '@mui/icons-material/Close';
+import useMediaQuery from '@mui/material/useMediaQuery';
 
 
-const settings = ['Profile', 'Account', 'Dashboard', 'Logout'];
-
 const Header:React.FC = () => {
-  const [anchorElNav, setAnchorElNav] = React.useState(null);
-  const [anchorElUser, setAnchorElUser] = React.useState(null);
-  const [mobileMoreAnchorEl, setMobileMoreAnchorEl] = React.useState<null | HTMLElement>(null);
+  const location = useLocation();
+  const [anchorElNav, setAnchorElNav] = useState(null);
+  const [mobileMoreAnchorEl, setMobileMoreAnchorEl] = useState<null | HTMLElement>(null);
+  const user = useContext(UserContext);
+  const navigate = useNavigate();
+  const [isUserAuthorized, setUserAuthorized] = useState(false);
+  const isMobileDevice = useMediaQuery('(min-width:900px)');
 
-  const handleOpenUserMenu = (event:any) => {
-    setAnchorElUser(event.currentTarget);
-  };
+  React.useEffect(() => {
+    if (user?.isLoggedIn) {
+      setUserAuthorized(true)
+    } else if (!user?.isLoggedIn) {
+      setUserAuthorized(false)
+    }
```

Comment:
Seems like this can be replaced with
```
setUserAuthorized(user?.isLoggedIn)
```

Reply:
I am not sure, how can I fix that, as my user can be null, so if I remove the check, I receive such error: `Argument of type 'boolean | undefined' is not assignable to parameter of type 'SetStateAction<boolean>'.
  Type 'undefined' is not assignable to type 'SetStateAction<boolean>'`